### PR TITLE
Add replica set detail page and refactor backend

### PR DESCRIPTION
**Pull request by floreks**
This PR has grown quite big but this is due to tests :).

I've refactored backend code related to events and used resource channels. On frontend I've extracted events to `kdEventCardList` component and prepared basic view for replica set details.

zrzut ekranu z 2016-05-11 14-21-16
zrzut ekranu z 2016-05-11 14-21-22

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/dashboard/730)

### CODE REVIEW

**Comment by floreks** [comment]
@digitalfishpond and @bryk could you review?

File: src/app/backend/resource/event/eventcommon.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import (
+	"github.com/kubernetes/dashboard/resource/common"
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+// GetEvents gets events associated to resource with given name.
+func GetEvents(client client.Interface, namespace, resourceName string) ([]api.Event, error) {
```
Comment (bryk) [inline src/app/backend/resource/event/eventcommon.go]: I believe you can narrow this to client.EventInterface

File: src/app/backend/resource/event/eventcommon.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+	eventList := <-channels.EventList.List
+	if err := <-channels.EventList.Error; err != nil {
+		return nil, err
+	}
+
+	return eventList.Items, nil
+}
+
+func GetPodsEvents(client client.Interface, namespace string, resourceSelector map[string]string) (
```
Comment (bryk) [inline src/app/backend/resource/event/eventcommon.go]: Can you document this?

**Comment by bryk** [comment]
That's a big one :) Two comments and LGTM. Please merge when done.

**Comment by floreks** [comment]
Review status: 0 of 43 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/app/backend/resource/event/eventcommon.go, line 26 \[r2\]](https://reviewable.io:443/reviews/kubernetes/dashboard/730#-KHUl6Fl8WrTIwWPTbSN-r2-26:-KHV2cGXgyA4Im2mn-EH:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/aba0224d8b996199beae925ff7beec0ac474feee/src/app/backend/resource/event/eventcommon.go#L26)):*
<details><summary><i>Previously, bryk (Piotr Bryk) wrote…</i></summary>
> I believe you can narrow this to client.EventInterface or EventNamespacer</details>
Done.

---

*[src/app/backend/resource/event/eventcommon.go, line 50 \[r2\]](https://reviewable.io:443/reviews/kubernetes/dashboard/730#-KHUl6Fl8WrTIwWPTbSN-r2-50:-KHV2c-ynPQFbUvE6jC-:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/aba0224d8b996199beae925ff7beec0ac474feee/src/app/backend/resource/event/eventcommon.go#L50)):*
<details><summary><i>Previously, bryk (Piotr Bryk) wrote…</i></summary>
> Can you document this? </details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/730)*

**Pull request merged**